guides/get-started-microservices-on-kubernetes/simple-microservice-client-part2.adoc

```diff
@@ -0,0 +1,293 @@
+
+Tag the Docker image:
+
+[source,bash,subs="normal"]
+----
+podman tag {my-jaxrs-app-docker-image-name-server} quay.io/{quay-io-account-name}/{my-jaxrs-app-docker-image-name-server}
+----
+
+Push the `{my-jaxrs-app-docker-image-name-server}` Docker Image to it:
+
+[source,bash,subs="normal"]
+----
+podman push quay.io/{quay-io-account-name}/{my-jaxrs-app-docker-image-name-server}
+----
+
+At this point, the `{my-jaxrs-app-docker-image-name-server}:latest` Docker Image should be publicly available and free to be consumed by any Kubernetes Cluster;
+
+=== Deploy on Kubernetes
+
+Create a file named `{my-jaxrs-app-docker-image-name-server}-deployment.yaml`:
+
+.{my-jaxrs-app-docker-image-name-server}-deployment.yaml:
+[source,yaml,subs="normal"]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {my-jaxrs-app-docker-image-name-server}-deployment
+  labels:
+    app: {my-jaxrs-app-docker-image-name-server}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {my-jaxrs-app-docker-image-name-server}
+  template:
+    metadata:
+      labels:
+        app: {my-jaxrs-app-docker-image-name-server}
+    spec:
+      containers:
+        - name: {my-jaxrs-app-docker-image-name-server}
+          image: quay.io/tborgato/{my-jaxrs-app-docker-image-name-server}
+          ports:
+            - containerPort: 8080
+            - containerPort: 9990
+          livenessProbe:
+            httpGet:
+              path: /health/live
+              port: 9990
+          readinessProbe:
+            httpGet:
+              path: /health/ready
+              port: 9990
+          startupProbe:
+            httpGet:
+              path: /health/started
+              port: 9990
+----
+
+Deploy to your Kubernetes Cluster:
+
+[source,bash,subs="normal"]
+----
+kubectl apply -f {my-jaxrs-app-docker-image-name-server}-deployment.yaml
+----
+
+=== Create Kubernetes ClusterIP Service
+
+We create a service to consume the services exposed by **{my-jaxrs-app-docker-image-name-server}** from inside Kubernetes;
+
+Create a file named `{my-jaxrs-app-docker-image-name-server}-service.yaml`:
+
+.{my-jaxrs-app-docker-image-name-server}-service.yaml:
+[source,yaml,subs="normal"]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: {my-jaxrs-app-docker-image-name-server}-service
+  labels:
+    app: {my-jaxrs-app-docker-image-name-server}
+spec:
+  ports:
+    - name: http
+      protocol: TCP
+      port: 8080
+      targetPort: 8080
+  selector:
+    app: {my-jaxrs-app-docker-image-name-server}
+  type: ClusterIP
+----
+
+Deploy to your Kubernetes Cluster:
+
+[source,bash,subs="normal"]
+----
+kubectl apply -f {my-jaxrs-app-docker-image-name-server}-service.yaml
+----
+
+=== Check Kubernetes Service
+
+[source,bash,subs="normal"]
+----
+kubectl run --rm -it --tty curl-{my-jaxrs-app-docker-image-name-server} --image=curlimages/curl --restart=Never &dash;&dash; {my-jaxrs-app-docker-image-name-server}-service:8080/hello/pippo
```

Comment:
This doesn't work for me. It's possible I don't have `minikube` started correctly. I just did `minikube start`.

```
If you don't see a command prompt, try pressing enter.
curl: (6) Could not resolve host: xxx
Hello 'pippo'.pod "curl-my-jaxrs-app-server" deleted
```

Reply:
I just deleted and re-installed a fresh minikube .... perhaps the two dashed get copied to some other character ... 

try:
```shell
kubectl run --rm -it --tty curl-simple-microservice-server --image=curlimages/curl --restart=Never -- http://simple-microservice-server-service:8080/hello/pippo
```

it worked for me ....